server/leafnode.go

```diff
@@ -125,6 +125,58 @@ func validateLeafNode(o *Options) error {
 	if err := validateLeafNodeAuthOptions(o); err != nil {
 		return err
 	}
+	// In local config mode, check that leafnode configuration refers to accounts that exist.
+	if len(o.TrustedOperators) == 0 {
```

Comment:
I was referring to moving the `else` part that you introduced, not the whole thing. I see that you had to replace the s.accounts.Load() to iterate through the options account.. and will have to do that for LN's account, possibly every LN user, every remote's local account.

Reply:
that's what I meant by move all checks. move it back?

Reply:
Sorry for the misunderstanding. Yes, I would feel more comfortable leaving the non operator mode checks where they were. You can have the ones specific to operator mode in validate options. But even then, if you feel that there may be some initialization not done at the time we call validate, then leave the check where you originally had them.
Tthe one I have in mind is this:
```
	if !s.processTrustedKeys() {
		return nil, fmt.Errorf("Error processing trusted operator keys")
	}

	s.mu.Lock()
	defer s.mu.Unlock()
        ...
```
We call validate before call to s.processTrustedKeys(). Maybe we can move call to validateOptions() after that call?

Reply:
I'll create a map of account names on the fly, shouldn't be to bad, considering we build it up later as well

Reply:
But if you move it back, you don't need that since you can access s.accounts.Load().. Basically undo your changes and just move the new checks into validate()

Reply:
the advantage would be that validation code stays together, wouldn't you agree?

Reply:
I would agree... but just want to make sure that we are not missing things that may be updated somewhere during the NewServer() init and before the original test was located.

Reply:
in this part we only check if accounts are configured, so far we generate only $G and $SYS